dhis-2/dhis-services/dhis-service-node/src/main/java/org/hisp/dhis/fieldfilter/DefaultFieldFilterService.java

```diff
@@ -429,11 +424,22 @@ else if ( property.isIdentifiableObject() && isProperIdObject( propertyClass ) )
                     child = new CollectionNode( property.getCollectionName() );
                     child.setNamespace( property.getNamespace() );
 
-                    for ( Object collectionObject : (Collection<?>) returnValue )
+                    for ( Object collectionObject : (Collection<?>) Objects.requireNonNull( returnValue ) )
                     {
```

Comment:
Maybe add a little message to this possible require non null, I guess this is not supposed to happen, but if a small explanation what is possible wrong could be nice, will also work as a sort of documentation possibly.

Reply:
Yeah, it was basically to get rid of a IDE warning.. it has been working for the last 6 years :) there is no reason it would ever be null.